File: code/community/1306330637468176395/deepgram-diarization-pricing.py
import os
import requests

api_key = os.getenv("DEEPGRAM_API_KEY")
audio_url = os.getenv("AUDIO_URL")

headers = {
    "Authorization": f"Token {api_key}",
    "Content-Type": "application/json"
}

payload = {
    "url": audio_url,
    "diarize": True
}

response = requests.post("https://api.deepgram.com/v1/listen", headers=headers, json=payload)

if response.status_code == 200:
    print("Transcription result:", response.json())
else:
    print("Error:", response.status_code, response.text)

File: code/community/1306330637468176395/deepgram-diarization-pricing.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

class Program
{
    private static readonly HttpClient client = new HttpClient();

    static async Task Main()
    {
        var apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        var audioUrl = Environment.GetEnvironmentVariable("AUDIO_URL");

        client.DefaultRequestHeaders.Add("Authorization", $"Token {apiKey}");

        var payload = new
        {
            url = audioUrl,
            diarize = true
        };

        var content = new StringContent(JsonConvert.SerializeObject(payload), Encoding.UTF8, "application/json");

        var response = await client.PostAsync("https://api.deepgram.com/v1/listen", content);

        var responseString = await response.Content.ReadAsStringAsync();

        Console.WriteLine(response.IsSuccessStatusCode ? "Transcription result: " + responseString : "Error: " + response.StatusCode + responseString);
    }
}